app/controllers/DatasetController.scala

```diff
@@ -69,6 +69,13 @@ object SegmentAnythingMaskParameters {
   implicit val jsonFormat: Format[SegmentAnythingMaskParameters] = Json.format[SegmentAnythingMaskParameters]
 }
 
+trait MetadataAssertions {
+  def assertNoDuplicateMetadataKeys(metadata: JsArray)(implicit ec: ExecutionContext, provider: MessagesProvider): Fox[Unit] = {
+    val keys = metadata.value.flatMap(_.as[JsObject] \\ "key").map(_.as[String]).toList
+    if (keys.size == keys.distinct.size) Fox.successful(()) else Fox.failure(Messages("dataset.metadata.duplicateKeys"))
+  }
+}
+
```

Comment:
Maybe you know a better place where to put this in the code base. 

I figured that a trait is better to not have to implement `assertNoDuplicateMetadataKeys` in both controllers of dataset and folder

Reply:
Ok I put it into  `app/utils` into an own file :)